duecredit/collector.py

```diff
@@ -259,8 +258,21 @@ def cite(self, entry, **kwargs):
                 version = external_versions[package]
             citation.version = version
 
+        self.citations[CitationKey(path, entry_key)] = citation
+
         return citation
 
+    def citations_fromentrykey(self):
+        """Return a dictionary with the current citations indexed by the entry key"""
+        citations = self.citations
+        citations_key = dict()
+        for (path, entry_key), citation in iteritems(citations):
```

Comment:
minor: since citations was used only once, there were no reason to assign it to some local variable